src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -303,6 +309,22 @@ private AsyncMethodHelper AsyncMethodHelper
             }
         }
 
+        private Assembly targetAssembly = null;
+        private bool tryToLoadAssembly = true;
+        private Assembly TargetAssembly
+        {
+            get
+            {
+                if (tryToLoadAssembly)
+                {
+                    targetAssembly = Assembly.LoadFrom(sourceAssembly);
```

Comment:
You're right; I didn't notice AsyncMethodHelper was in another AppDomain probably because I looked at a narrow scope. In that case AsyncMethodHelper.LoadAssembly should be restored back and the lookup code for defining types should be rather moved to another(new) method inside AsyncMethodHelper.